torch/csrc/tensor/python_tensor.cpp

```diff
@@ -327,54 +348,30 @@ void py_set_default_tensor_type(PyObject* obj) {
   } else {
     throw TypeError("invalid type object");
   }
-  auto aten_type = type->aten_type();
-  auto scalar_type = static_cast<ScalarType>(type->scalar_type);
-  if (!aten_type) {
-    throw unavailable_type(*type);
-  }
-  set_default_tensor_type(*aten_type, scalar_type);
+  set_default_tensor_type(type);
```

Comment:
I don't know if this is due to this specific change or somewhere else, but this seems worse:

Before you patch (without CUDA):
```
torch.set_default_tensor_type(torch.cuda.FloatTensor)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-2-876b829d82b3> in <module>()
----> 1 torch.set_default_tensor_type(torch.cuda.FloatTensor)

/data/users/gchanan/_pytorch11/torch/__init__.py in set_default_tensor_type(t)
    151     if isinstance(t, _string_classes):
    152         t = _import_dotted_name(t)
--> 153     _C._set_default_tensor_type(t)
    154
    155

/data/users/gchanan/_pytorch11/torch/cuda/__init__.py in _lazy_init()
    176         raise RuntimeError(
    177             "Cannot re-initialize CUDA in forked subprocess. " + msg)
--> 178     _check_driver()
    179     torch._C._cuda_init()
    180     _cudart = _load_cudart()

/data/users/gchanan/_pytorch11/torch/cuda/__init__.py in _check_driver()
     90 def _check_driver():
     91     if not hasattr(torch._C, '_cuda_isDriverSufficient'):
---> 92         raise AssertionError("Torch not compiled with CUDA enabled")
     93     if not torch._C._cuda_isDriverSufficient():
     94         if torch._C._cuda_getDriverVersion() == 0:

AssertionError: Torch not compiled with CUDA enabled
```

with your patch:
```
In [4]: torch.set_default_tensor_type(torch.cuda.FloatTensor)

In [5]: torch.randn(2,3)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-5-1dc2fcf8de67> in <module>()
----> 1 torch.randn(2,3)

/data/users/gchanan/_pytorch8/torch/cuda/__init__.py in _lazy_init()
    176         raise RuntimeError(
    177             "Cannot re-initialize CUDA in forked subprocess. " + msg)
--> 178     _check_driver()
    179     torch._C._cuda_init()
    180     _cudart = _load_cudart()

/data/users/gchanan/_pytorch8/torch/cuda/__init__.py in _check_driver()
     90 def _check_driver():
     91     if not hasattr(torch._C, '_cuda_isDriverSufficient'):
---> 92         raise AssertionError("Torch not compiled with CUDA enabled")
     93     if not torch._C._cuda_isDriverSufficient():
     94         if torch._C._cuda_getDriverVersion() == 0:

AssertionError: Torch not compiled with CUDA enabled
```

Reply:
fixed and added a test for it.